src/Symfony/Component/DependencyInjection/Debug/AutowiringInfoManager.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Debug;
+
+/**
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+final class AutowiringInfoManager
+{
+    /**
+     * @var AutowiringInfoProviderInterface[]
+     */
+    private $autowiringInfoProviders;
+
+    private $typeInfos = null;
+
+    public function __construct(array $autowiringInfoProviders)
+    {
+        $this->autowiringInfoProviders = $autowiringInfoProviders;
+    }
+
+    /**
+     * @param string $type
+     *
+     * @return AutowiringTypeInfo|null
+     */
+    public function getInfo(string $type)
```

Comment:
`: ?AutowiringTypeInfo` (docblock is redundant)

Reply:
+1 I keep thinking Symfony requires 7.0, so no nullable types :). I'll change this happily